src/hotspot/cpu/x86/x86.ad

```diff
@@ -7192,9 +7252,9 @@ instruct vcmpu32(legVec dst, legVec src1, legVec src2, immI8 cond, legVec vtmp1,
 
 instruct vcmpu64(vec dst, vec src1, vec src2, immI8 cond, rRegP scratch, kReg ktmp) %{
   predicate(UseAVX > 2 &&
-            !VM_Version::supports_avx512vl() &&
-            vector_length_in_bytes(n->in(1)->in(1)) == 64 && // src1
-            is_integral_type(vector_element_basic_type(n->in(1)->in(1)))); // src1
+            (VM_Version::supports_avx512vl() ||
+             Matcher::vector_length_in_bytes(n->in(1)->in(1)) == 64) && // src1
```

Comment:
This doesn't look correct. This instruct sets vec dst, whereas with avx512vl kReg dst is expected.
Should have been:
            !VM_Version::supports_avx512vl() &&
            vector_length_in_bytes(n->in(1)->in(1)) == 64 &&

Reply:
Yes this new pattern was added specifically for 64 byte unsigned/signed comparisons (will apply to integral types only) for targets which do not support VL extensions.